dotnet/packages/Microsoft.Bot.Builder.M365/Microsoft.Bot.Builder.M365/AI/Moderator/OpenAIModerator.cs

```diff
@@ -41,13 +42,22 @@ public OpenAIModerator(OpenAIModeratorOptions options, ILogger? logger = null, H
             {
                 case ModerationType.Input:
                 case ModerationType.Both:
-                    {
-                        // get input from turnstate
-                        // TODO: when TurnState is implemented, get the user input
-                        string input = turnContext.Activity.Text;
+                {
+                    string input = turnState.TempState?.Value.Input ?? turnContext.Activity.Text;
+
+                    // TODO: Refactor turn state to fix convoluted logic
 
-                        return await _HandleTextModeration(input, true);
+                    // Get input from turn state
+                    if (turnState as object is TurnState defaultTurnState)
+                    {
+                        if (defaultTurnState.TempState?.Value.Input != null)
+                        {
+                            input = defaultTurnState.TempState?.Value.Input!;
```

Comment:
Just remind that `AzureContentSafetyModerator` does not include this part. Is it by design?

Reply:
Added it, thanks for calling out